lib/rubygems/ext/rake_builder.rb

```diff
@@ -9,7 +9,7 @@ class Gem::Ext::RakeBuilder < Gem::Ext::Builder
 
   def self.build(extension, directory, dest_path, results, args=[], lib_dir=nil)
     if File.basename(extension) =~ /mkrf_conf/i then
-      cmd = "#{Gem.ruby} #{File.basename extension}"
+      cmd = "#{Gem.ruby} #{File.basename extension}".dup
       cmd << " #{args.join " "}" unless args.empty?
```

Comment:
Rewriting `" #{args.join " "}"` as `" #{args.join(" ")}"` is clearer to reader there's a string param passed into `join` and it's not `" #{args.join "` +  `"}"`. Maybe use single quotes to be more obvious.